src/NLog.Targets.Redis/RedisTarget.cs

```diff
@@ -1,108 +1,112 @@
-        /// </summary>
-        public string Password { get; set; }
-
-        private RedisConnectionManager _redisConnectionManager;
-
-        internal virtual RedisConnectionManager CreateConnectionManager()
-        {
-            return new RedisConnectionManager(Host, Port, Db, Password);
-        }
-
-        protected override void InitializeTarget()
-        {
-            _dataTypeToLower = DataType?.ToLower();
-            _redisConnectionManager = CreateConnectionManager();
-            _redisConnectionManager.InitializeConnection();
-
-            base.InitializeTarget();
-        }
-
-        protected override void CloseTarget()
-        {
-            if (_redisConnectionManager != null)
-            {
-                _redisConnectionManager.Dispose();
-            }
-
-            base.CloseTarget();
-        }
-
-        protected override void Write(LogEventInfo logEvent)
-        {
-            var message = this.Layout.Render(logEvent);
-            var key = _key?.Render(logEvent);
-            var redisDatabase = _redisConnectionManager.GetDatabase();
-            switch (_dataTypeToLower)
-            {
-                case ListDataType:
-                    redisDatabase.ListRightPush(key, message);
-                    break;
-                case ChannelDataType:
-                    redisDatabase.Publish(key, message);
-                    break;
-                default:
-                    throw new Exception("no data type defined for redis");
-            }
-        }
-    }
+﻿using System;
+using NLog.Config;
+using NLog.Layouts;
+
+namespace NLog.Targets.Redis
+{
+    [Target("Redis")]
+    public class RedisTarget : TargetWithLayout
+    {
+        protected const string ListDataType = "list";
+        protected const string ChannelDataType = "channel";
+
+        /// <summary>
+        /// Sets the host name or IP Address of the redis server
+        /// </summary>
+        [RequiredParameter]
+        public string Host { get; set; }
+
+        /// <summary>
+        /// Sets the port number redis is running on
+        /// </summary>
+        [RequiredParameter]
+        public int Port { get; set; }
+
+        /// <summary>
+        /// Sets the key to be used for either the list or the pub/sub channel in redis
+        /// </summary>
+        [RequiredParameter]
+        public string Key
+        {
+            get
+            {
+                SimpleLayout simpleLayout = _key as SimpleLayout;
+                if (simpleLayout != null)
+                    return simpleLayout.Text;
+                else if (_key != null)
+                    return _key.ToString();
+                else
+                    return null;
+            }
+            set { _key = value; }
+        }
+
+        private Layout _key;
+
+        /// <summary>
+        /// Sets what redis data type to use, either "list" or "channel", defaults to "list"
+        /// </summary>
+        public string DataType { get; set; }
```

Comment:
maybe set `[DefaultValue]`, and maybe we should set it in the property as default? e.g.

```
public string DataType { get; set; } = ListDataType
```

PS I do prefer to describe all changes in the PR title.

Reply:
> Does NLog do anything special with the `[DefaultValue]`

Not really. It's a good habit I think

Reply:
> I also should have made a separate pull request for the backwards compatibility.

That's indeed better IMO, shorter (scoped) PRs are really preferred. Easier to review, discuss and approve :)